tests/ui/linkage-attr/unreferenced-used-static-issue-127052.rs

```diff
@@ -1,9 +1,9 @@
-// This is a non-regression test for issue #127052 where unreferenced `#[used]` statics couldn't be
-// removed by the MSVC linker, causing linking errors.
+// This is a non-regression test for issue #127052 where unreferenced `#[used]` statics in the
+// binary crate couldn't be found by the MSVC linker, causing linking errors.
```

Comment:
What do you think of the following?

```suggestion
// This is a non-regression test for issue #127052 where unreferenced `#[used]` statics
// were forcefully exported and included in the final binary, causing linking errors on MSVC.
```

Reply:
They were not included though, wasn't that the issue? If they had been included the linker would not have complained about missing symbols.

Reply:
I meant the symbol was added to the list of exported symbols.

Reply:
Actually maybe just steal https://github.com/rust-lang/rust/pull/126938#issuecomment-2192434302

Reply:
That is also confusing IMO, since "forcing them to be included in the final binary" sounds like what we want.

My understanding is that we are on the one hand telling the linker to export the symbol, but on the other hand the symbol is not actually included in the binary, and that leads to the error. But I may be misunderstanding, this is all very strange to me.

Cc @ChrisDenton

Reply:
Yeah basically. As far as I'm aware we don't have any tests that ensure `FOO` actually ends up in the final binary. It won't, at least on msvc, because we define it as a COMDAT (i.e. we tell the linker it's allowed to remove it).

This is technically in keeping with the [documented behaviour](https://doc.rust-lang.org/stable/reference/abi.html?highlight=used#the-used-attribute) which only says the compiler considers it used, and not the linker. But I think that was intended as a hedge and we should still make a best effort to ensure `#[used]` means it's used.

Reply:
There's also `used(linker)` which I guess is meant to tell the linker to keep it -- I don't understand why that is a separate flag though. :shrug:

Reply:
> I meant the symbol was added to the list of exported symbols.

So what about the current wording?

Reply:
whatever Chris wants, r? @ChrisDenton

Reply:
I'm finding it a bit hard to summarise without giving a full tutorial on linking. But this explanation seems accurate enough and the issue link helps if anyone wants to explore further. So, yep, this looks good to me.